state/core.go

```diff
@@ -199,6 +227,71 @@ func (c *Core) Height(commit flow.StateCommitment) (uint64, error) {
 	return height, nil
 }
 
+func (c *Core) Events(height uint64, types ...string) ([]flow.Event, error) {
+	// Make sure that the request is for a height below the currently active
+	// sentinel height; otherwise, we haven't indexed yet and we might return
+	// false information.
+	if height > c.height {
+		return nil, fmt.Errorf("unknown height (current: %d, requested: %d)", c.height, height)
+	}
+
+	lookup := make(map[uint64]struct{})
+	for _, typ := range types {
+		lookup[xxhash.Checksum64([]byte(typ))] = struct{}{}
+	}
+
+	// Iterate over all keys within the events index which are prefixed with the right block height.
+	var events []flow.Event
+	prefix := make([]byte, 1+8)
+	prefix[0] = model.PrefixEventIndex
+	binary.BigEndian.PutUint64(prefix[1:], height)
+	err := c.index.View(func(tx *badger.Txn) error {
+		it := tx.NewIterator(badger.IteratorOptions{
+			Prefix: prefix,
+		})
+		defer it.Close()
+
+		// Iterate on all keys with the right prefix.
+		for it.Seek(prefix); it.ValidForPrefix(prefix); it.Next() {
+			item := it.Item()
+			key := item.Key()
+
+			// If types were given for filtering, discard events which should not be included.
+			hash := binary.BigEndian.Uint64(key[1+8:])
+			_, ok := lookup[hash]
+			if len(lookup) != 0 && !ok {
+				continue
+			}
+
+			// Unmarshal event batch and append them to result slice.
+			var evts []flow.Event
```

Comment:
I just have to say, why save two characters here? XD

Reply:
A wise man with a keyboard with no vowels once taught me that vowels in variable names were a mere distraction and that the real essence of names lies in their consonants! 🙏 

Jokes aside though, we already have an `events` variable and it was just long enough (18 lines) that I thought calling it `e` might be a bit optimistic, so I applied the aforementioned venerable wisdom and removed the vowels! hehe

Reply:
Fair enough :D

We could use `ee` here, if we wanted to stay consistent with some other code, but really irrelevant.

Reply:
Let's leave it like this right now then and if/when we write a style guide for our team this can be part of a style-fixing PR :p